### PR TITLE
Use str.format()

### PR DESCRIPTION
**Pull request by whyscream**
Addresses https://github.com/mollie/mollie-api-python/pull/45#discussion_r208542381

Merge #52 first please.

### CODE REVIEW

File: examples/4-ideal-payment.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def main():
         if 'issuer' not in flask.request.form:
             body = '<form method="post">Select your bank: <select name="issuer">'
             for issuer in mollie_client.methods.get('ideal', include='issuers').issuers:
-                body += '<option value="%s">%s</option>' % (issuer.id, issuer.name)
+                body += '<option value="%{id}">{issuer}</option>'.format(id=issuer.id, issuer=issuer.name)
```
Comment (mvdpanne) [inline examples/4-ideal-payment.py]: Leftover `%` here?

File: mollie/api/resources/base.py [hunk 2]
```diff
@@ -60,29 +60,30 @@ def update(self, resource_id, data=None, **params):
...
         return self.rest_list(params)
 
     def perform_api_call(self, http_method, path, data=None, params=None):
         resp = self.client.perform_http_call(http_method, path, data, params)
         try:
             result = resp.json() if resp.status_code != 204 else {}
         except Exception:
-            raise ResponseHandlingError('Unable to decode Mollie API response (status code: %d): %s' % (
-                resp.status_code, resp.text))
+            raise ResponseHandlingError("Unable to decode Mollie API response "
```
Comment (mvdpanne) [inline mollie/api/resources/base.py]: To me, this seems more readable:
```
raise ResponseHandlingError(
    "Unable to decode Mollie API response (status code: {status}): '{response}'."
    .format(status=resp.status_code, response=resp.text)
)
```

File: mollie/api/resources/base.py [hunk 3]
```diff
@@ -60,29 +60,30 @@ def update(self, resource_id, data=None, **params):
...
+                                            status=resp.status_code, response=resp.text))
         if resp.status_code < 200 or resp.status_code > 299:
             if 'status' in result and (result['status'] < 200 or result['status'] > 299):
                 # the factory will return the appropriate ResponseError subclass based on the result
                 raise ResponseError.factory(result)
             else:
-                raise ResponseHandlingError(
-                    'Received HTTP error from Mollie API, but no status in payload (status code: %d): %s' % (
-                        resp.status_code, resp.text))
+                raise ResponseHandlingError("Received HTTP error from Mollie API, but no status in "
```
Comment (mvdpanne) [inline mollie/api/resources/base.py]: Same here:
```
raise ResponseHandlingError(
    "Received HTTP error from Mollie API, but no status"
    " in payload (status code: {status}): '{response}'."
    .format(status=resp.status_code, response=resp.text)
)
```

File: mollie/api/resources/customer_subscriptions.py [hunk 4]
```diff
@@ -8,15 +8,13 @@ class CustomerSubscriptions(Base):
...
-        subscription._resource = self
-        return subscription
+        return Subscription(result, self)
 
     def get(self, subscription_id, **params):
         if not subscription_id or not subscription_id.startswith(self.RESOURCE_ID_PREFIX):
             raise IdentifierError(
-                'Invalid subscription ID: "%s". A subscription ID should start with "%s".' % (
-                    subscription_id, self.RESOURCE_ID_PREFIX)
+                "Invalid subscription ID: '{id}'. A subscription ID should start with '{prefix}'.".format(
```
Comment (mvdpanne) [inline mollie/api/resources/customer_subscriptions.py]: I'd prefer the `.format()` on a new line.

Comment (whyscream) [reply mollie/api/resources/customer_subscriptions.py]: We currently do this everywhere in the code: if we need to wrap code (>119 chars) and we can do it on an open parenthese, we use that. Changing that in a few places is more disturbing IMHO.

Comment (mvdpanne) [reply mollie/api/resources/customer_subscriptions.py]: Alright, fair.

**Comment by mvdpanne** [comment]
If you could rebase the branch, I will merge it ;)

**Comment by whyscream** [comment]
Rebased :)

**Pull request merged**